plugins/modules/custom_images.py

```diff
@@ -0,0 +1,315 @@
+#
+# Copyright (c) 2021, Ciril Troxler<ciril.troxler@cloudscale.ch>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: custom_images
+short_description: Manage custom images on the cloudscale.ch IaaS service
+description:
+  - Import, modify and delete Custom images.
+notes:
+  - To import a new custom-image the I(url) and I(name) options are required.
+author:
+  - Ciril Troxler (@ctx)
+version_added: 2.2.0
+options:
+  url:
+    description:
+      - The URL used to download the image.
+    type: str
+  name:
+    description:
+      - The human readable name of the custom image. Either name or UUID must
+        be present to change an existing image.
+    type: str
+  uuid:
+    description:
+      - The unique identifier of the custom image import. Either name or UUID
+        must be present to change an existing image
+    type: str
+  slug:
+    description:
+      - A string identifying the custom image for use within the API.
+    type: str
+  user_data_handling:
+    description:
+      - How user_data will be handled when creating a server. There are
+        currently two options, "pass-through" and "extend-cloud-config".
+    type: str
+    choices: [ pass-through , extend-cloud-config ]
+  zones:
+    description:
+      - Specify zones in which the custom image will be available (e.g. C(lpg1)
+        or C(rma1)).
+    type: list
+    elements: str
+  source_format:
+    description:
+      - The file format of the image referenced in the url. Currently only raw
+        is supported.
+    default: "raw"
+    choices: [ raw ]
+    type: str
+  tags:
+    description:
+      - The tags assigned to the custom image.
+    type: dict
+  state:
+    description: State of the coustom image.
+    choices: [ present, absent ]
+    default: present
+    type: str
+  importstatus:
+    description: Get the import status of a new custom image.
+    type: bool
+    choices: [ true, false]
+    default: false
+  force:
+    description: Import an image
+    choices: [ true, false ]
+    default: false
+    type: bool
+extends_documentation_fragment: cloudscale_ch.cloud.api_parameters
+'''
+
+EXAMPLES = r'''
+- name: Import custom image
+  cloudscale_ch.cloud.custom_images:
+    name: my-custom-image
+    url: https://ubuntu.com/downloads/hirsute.img
+    slug: SLUG
+    user_data_handling: pass-throug
+    zones: LPG1
+    tags:
+      project: luna
+    state: present
+
+- name: Wait until import succeeded
+  cloudscale_ch.cloud.custom_images:
+    uuid: 11111111-1864-4608-853a-0771b6885a3a
+    importstatus: true
+  retries: 15
+  delay: 5
+  register: importstatus
+  until: importstatus.status == 'success'
+
+- name: Update custom image
+  cloudscale_ch.cloud.custom_images:
+    name: my-custom-image
+    slug: SLUG
+    user_data_handling: pass-throug
```

Comment:
In most cases, `extend-cloud-config` is more likely to be what the user wants. So I would use that in these examples.